### PR TITLE
added SonarQube reporter.

### PR DESCRIPTION
**Pull request by junkpiano**
code for #2350 .

### CODE REVIEW

File: Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public static let identifier = "sonarqube"
+    public static let isRealtime = false
+
+    public var description: String {
+        return "Reports violations in SonarQube import format."
+    }
+
+    public static func generateReport(_ violations: [StyleViolation]) -> String {
+        return toJSON(["issues": violations
+            .filter { $0.location.relativeFile != nil && $0.location.line != nil }
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift]: Is there a reason to filter? We already have to use optional chaining below when creating the dictionary

File: Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            .map(dictionary(for:))
+            ])
+    }
+
+    // refer to https://docs.sonarqube.org/display/SONAR/Generic+Issue+Data
+    static func dictionary(for violation: StyleViolation) -> [String: Any] {
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift]: `private`?

Comment (junkpiano) [reply Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift]: yes. I modified.

File: Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift (hunk 1 above)
Comment (junkpiano) [reply Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift]: Exactly. It was kind of error handling, but I tried again without filtering, and I found it works. so I removed this filtering.

**Comment by junkpiano** [comment]
@marcelofabri updated. and rebased on the lastest master to resolve conflicts.

**Pull request merged**